### PR TITLE
Add new call recording VBC APIs

### PR DESCRIPTION
**Pull request by marklewin**
# Description

Adding new Call Recording APIs for an API review. Marking as "Don't Merge" because they are not yet production APIs. Leaving changelog for now because don't want to advertise these!

# Checklist

- [ ] version number incremented (in the `info` section of the spec)

### CODE REVIEW

File: definitions/vonage-business-cloud/odcr.yml [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+      name: cnam
+      in: query
+      description: Filter records by CNAM
+      required: false
+      schema:
+        type: string
+    Dnis:
+      name: dnis
+      in: query
+      description: Filter records by CNAM
```
Comment (mheap) [inline definitions/vonage-business-cloud/odcr.yml]: This description looks wrong. What is `dnis` short for?

File: definitions/vonage-business-cloud/ccr.yml [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+        page_size:
+          type: integer
+        total_items:
+          type: integer
+        total_pages:
+          type: integer
+    ExportJobHalResponse:
+      type: object
+      properties:
+        _embedded:
```
Comment (mheap) [inline definitions/vonage-business-cloud/ccr.yml]: As this is a single resource it shouldn't be `_embedded`

File: definitions/vonage-business-cloud/ccr.yml [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+          description: Not Found
+    delete:
+      operationId: deleteCCRecording
+      summary: Delete company call recording
+      description: Delete a single company call recording for an account
+      parameters:
+        - $ref: "#/components/parameters/AccountID"
+        - $ref: "#/components/parameters/RecordingID"
+      responses:
+        "200":
```
Comment (mheap) [inline definitions/vonage-business-cloud/ccr.yml]: Doesn't need `200` response if there's a `204`

File: definitions/vonage-business-cloud/odcr.yml [hunk 4]
```diff
@@ -0,0 +1,541 @@
...
+            type: integer
+        start:
+          type: string
+          format: date-time
+          description: Start time of the recorded call
+          example: "2019-01-01T00:00:00+0000"
+    RecordingHalResponse:
+      type: object
+      properties:
+        _embedded:
```
Comment (mheap) [inline definitions/vonage-business-cloud/odcr.yml]: As this is a single recording it doesn't need `_embedded`

File: definitions/vonage-business-cloud/ccr.yml [hunk 5]
```diff
@@ -0,0 +1,527 @@
...
+        "401":
+          description: Unauthorized
+        "403":
+          description: Forbidden
+        "404":
+          description: Not Found
+    delete:
+      operationId: deleteCCRecordings
+      summary: Delete multiple company call recordings
+      description: Delete multiple company call recordings for an account
```
[the remark this answers is not in the document]
Comment (bdorry) [reply definitions/vonage-business-cloud/ccr.yml]: Use case is clearing out recordings based on criteria, ex: recordings older than X

File: definitions/vonage-business-cloud/ccr.yml [hunk 6]
```diff
@@ -0,0 +1,527 @@
...
+          type: integer
+        total_items:
+          type: integer
+        total_pages:
+          type: integer
+    CreateExportJob:
+      type: object
+      properties:
+        job_id:
+          type: array
```
[the remark this answers is not in the document]
Comment (bdorry) [reply definitions/vonage-business-cloud/ccr.yml]: Depending on export size, yes there may be multiple chunks/jobs

File: definitions/vonage-business-cloud/ccr.yml [hunk 7]
```diff
@@ -0,0 +1,527 @@
...
+          items:
+            type: string
+          description: Unique identifier of the created export jobs
+          example:
+            - afa725fb-d418-4eaf-b3f9-0e66396fafdc
+    CreateExportJobHalResponse:
+      type: object
+      properties:
+        _embedded:
```
[the remark this answers is not in the document]
Comment (bdorry) [reply definitions/vonage-business-cloud/ccr.yml]: Ok Im updating the swagger and requesting these changes

File: definitions/vonage-business-cloud/odcr.yml (hunk 1 above)
Comment (bdorry) [reply definitions/vonage-business-cloud/odcr.yml]: Dialed Number Identification Service

It identifies the dialed number of the recording

File: definitions/vonage-business-cloud/ccr.yml (hunk 2 above)
Comment (bdorry) [reply definitions/vonage-business-cloud/ccr.yml]: Ok Im updating the swagger and requesting these changes

File: definitions/vonage-business-cloud/ccr.yml [hunk 8]
```diff
@@ -0,0 +1,527 @@
...
+        - $ref: "#/components/parameters/CallerID"
+        - $ref: "#/components/parameters/Cnam"
+        - $ref: "#/components/parameters/Dnis"
+        - $ref: "#/components/parameters/DurationLte"
+        - $ref: "#/components/parameters/DurationGte"
+      responses:
+        "200":
+          description: OK
+          content:
+            "*/*":
```
[the remark this answers is not in the document]
Comment (bdorry) [reply definitions/vonage-business-cloud/ccr.yml]: Yes, updated

**Comment by mheap** [comment]
Closing as it doesn't seem to be picking up new commits. Will open a new PR

**Pull request closed without merging**